### DISCUSSION TITLE
Suppress logging for single endpoint for special usage (Kubernetes Liveness Probe)

### DISCUSSION THREAD

**Discussion by sebastianbuechler** [fastapi/fastapi#6998] [Questions] [answered]
I am using fastapi in a container that is deployed to Kubernetes. Kubernetes offers the feature of liveness and readiness probes which are basically just HTTP get calls watched for status code <400. In fastapi it is easy to just define such an endpoint like this:

```
@fastapiserver.get("/healthz")
# Liveness probe for kubernetes status service
def kubernetes_liveness_probe():
    return {"status": "healthy"}
```

However, since the probes call the endpoint quite often (~every 5 seconds) the log gets full of the calls. 

How can I suppress logging for a single endpoint without losing the logging for all the others?

**Comment by Kludex** [comment]
You need to add a filter on the handlers available for the `uvicorn.access` logger.

**Comment by sebastianbuechler** [comment, marked as answer]
Thank you for answering, the hint to the filtering method led me to [here](https://github.com/encode/starlette/issues/864#issuecomment-653076434).

So if somebody else needs this behaviour, here is the solution of @HarrySky:
```python
class EndpointFilter(logging.Filter):
    def filter(self, record: logging.LogRecord) -> bool:
        return record.getMessage().find("/endpoint") == -1

# Filter out /endpoint
logging.getLogger("uvicorn.access").addFilter(EndpointFilter())
```

Where "/endpoint" is the endpoint you want to suppress logging for.

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @sebastianbuechler  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
